dashboard/app/dsl/script_dsl.rb

```diff
@@ -76,21 +78,43 @@ def level_concept_difficulty(json)
 
   string :video_key_for_next_level
 
+  boolean :active
+
   def assessment(name)
     level(name, {assessment: true})
   end
 
   def level(name, properties = {})
-    @levels << {
+    level = {
       :name => name,
-      :stage => @stage,
       :stage_flex_category => @stage_flex_category,
       :skin => @skin,
       :concepts => @concepts.join(','),
       :level_concept_difficulty => @level_concept_difficulty || {},
       :video_key => @video_key_for_next_level
     }.merge(properties).select{|_, v| v.present? }
     @video_key_for_next_level = nil
+    if @current_scriptlevel
+      @current_scriptlevel[:levels] << level
+      if !@active
+        @current_scriptlevel[:properties][name] = { :active => @active }
```

Comment:
Or actually, [`Store`](http://api.rubyonrails.org/classes/ActiveRecord/Store.html) might be the more 'official' way to apply pattern in Rails (I didn't know about `Store` when I initially wrote SerializedProperties, but it might be worth using `Store` for new/future applications of this pattern).